### DISCUSSION TITLE
Не загружаются видео на Youtube

### DISCUSSION THREAD

**Discussion by Wyrmlet** [hufrea/byedpi#218] [General]
Уже несколько дней сам ютуб работает, но видео не загружаются. Возможно какая-то новая блокировка для `googlevideo.com`? Возможно дубликат #190, но я не уверен.

**Comment by wassalam01** [comment]
Такая же проблема, причем стримы открываются, а видео нет. Еще заметил что днем более менее через раз, а то и каждый раз работает. А вот ночью как будто ресурсы дпи освобождаются из-за чего перестает работать, то что работало днем.

**Comment by radioxoma** [comment]
МГТС. Днём часто хватало `-s1` или `-d1`, вечером `-o1` или `-s1 -r1`. Сейчас `-s1 -r1` чинит часть сайта, но стрим через сайт или _yt-dlp_ не идёт. Установка соединения виснет. Ночью 23:00-04:00 всё совсем замирает.

Пробитые при помощи `-a1` UDP соединения, например wireguard, стали рваться после нескольких мегабайт передачи.

**Comment by Black-Semargl** [comment]
При закачке часть соединений проходит, часть нет, вот например

```
[download]  18.5% of   53.29MiB at  399.96KiB/s ETA 01:51
[download] Got error: SocksHTTPSConnectionPool(host='rr15---sn-axq7sn7l.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (1/10)...
[download]  36.9% of   53.29MiB at  399.65KiB/s ETA 01:26
[download] Got error: SocksHTTPSConnectionPool(host='rr2---sn-oxuctoxu-axqe.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (2/10)...
[download]  54.8% of   53.29MiB at  398.43KiB/s ETA 01:01
[download] Got error: SocksHTTPSConnectionPool(host='rr2---sn-oxuctoxu-axqe.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (3/10)...
[download] Got error: SocksHTTPSConnectionPool(host='rr2---sn-oxuctoxu-axqe.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (4/10)...
[download] Got error: SocksHTTPSConnectionPool(host='rr15---sn-axq7sn7l.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (5/10)...
[download]  73.4% of   53.29MiB at  388.76KiB/s ETA 00:37
[download] Got error: SocksHTTPSConnectionPool(host='rr15---sn-axq7sn7l.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (6/10)...
[download] Got error: SocksHTTPSConnectionPool(host='rr15---sn-axq7sn7l.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (7/10)...
[download]  91.3% of   53.29MiB at  399.93KiB/s ETA 00:11
[download] Got error: SocksHTTPSConnectionPool(host='rr15---sn-axq7sn7l.googlevideo.com', port=443): Read timed out. (read timeout=20.0). Retrying (8/10)...
[download] 100% of   53.29MiB in 00:05:04 at 179.45KiB/s

```

**Comment by H5117** [comment]
У меня на IPv6 работает так:
```
ciadpi --hosts youtube.txt --fake -2 --tls-sni "www.google.com" --ip-opt --tlsrec=6+s --tlsrec 27+s
```

**Comment by Miraihi** [comment]
> У меня на IPv6 работает так:
> 
> ```
> ciadpi --hosts youtube.txt --fake -2 --tls-sni "www.google.com" --ip-opt --tlsrec=6+s --tlsrec 27+s
> ```

Очень интересно что этот вариант работает лучше без выключения quic. `--tlsrec 27+s` это крайне эффективная опция (Не только делает --fake юзабельным, но и избавляет от задержки перед видео), не знал что ее имеет смысл выставлять так высоко.

**Comment by Yonnji** [comment]
> У меня на IPv6 работает так:
> 
> ```
> ciadpi --hosts youtube.txt --fake -2 --tls-sni "www.google.com" --ip-opt --tlsrec=6+s --tlsrec 27+s
> ```

Мне еще disorder понадобился, без него даже страницы не открывались. Получилось такое:
`--fake -2 --disorder 2 --tlsrec=6+s --tlsrec 27+s`

**Comment by H5117** [comment]
Похоже, я случайно обнаружил ещё один способ обхода блокировки `*.googlevideo.com`. Оказывается, некоторые сервера GGC корректно работают даже с неправильным SNI. Т. е. если тупо поменять SNI на `www.google.com` в ClientHello, сервер всё равно отвечает сертификатом `*.googlevideo.com`. Может кто-нибудь подтвердить?

<details> <summary>Эксперимент</summary>

```
$ openssl s_client -connect 74.125.163.97:443 -servername "www.google.com"
Connecting to 74.125.163.97
CONNECTED(00000003)
depth=2 C=US, O=Google Trust Services LLC, CN=GTS Root R1
verify return:1
depth=1 C=US, O=Google Trust Services, CN=WR2
verify return:1
depth=0 CN=*.googlevideo.com
verify return:1
---
Certificate chain
 0 s:CN=*.googlevideo.com
   i:C=US, O=Google Trust Services, CN=WR2
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Oct  8 14:31:06 2024 GMT; NotAfter: Dec 17 14:31:05 2024 GMT
 1 s:C=US, O=Google Trust Services, CN=WR2
   i:C=US, O=Google Trust Services LLC, CN=GTS Root R1
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Dec 13 09:00:00 2023 GMT; NotAfter: Feb 20 14:00:00 2029 GMT
 2 s:C=US, O=Google Trust Services LLC, CN=GTS Root R1
   i:C=BE, O=GlobalSign nv-sa, OU=Root CA, CN=GlobalSign Root CA
   a:PKEY: rsaEncryption, 4096 (bit); sigalg: RSA-SHA256
   v:NotBefore: Jun 19 00:00:42 2020 GMT; NotAfter: Jan 28 00:00:42 2028 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIGyTCCBbGgAwIBAgIRAIiaKa0lehC2CcuLTc82plUwDQYJKoZIhvcNAQELBQAw
OzELMAkGA1UEBhMCVVMxHjAcBgNVBAoTFUdvb2dsZSBUcnVzdCBTZXJ2aWNlczEM
MAoGA1UEAxMDV1IyMB4XDTI0MTAwODE0MzEwNloXDTI0MTIxNzE0MzEwNVowHDEa
MBgGA1UEAwwRKi5nb29nbGV2aWRlby5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IB
DwAwggEKAoIBAQCs9EuG9WWcDQ46fKeVBYW8yEbKrnEOtuZWkCnTm5WiszhH6Jp/
m2RcOCODRk29Xj7eoj3qDVOmKv/CNNnqtt5VF1p34r/REqpfQhYLWsLe5Ay2ODpB
7fg80akq9LRsomBfLQk5Pwc1L5G7JaqssILWlIdsAhxmAUAGydMgUs6HbLIO85FS
0TUhqZxGwDOrheyH5MXb0xmOf/q178A5Z90hhypDPH4SKR+9sYqL2DCSseF0oDzd
uACJkCdw1RPhGD2oOZ48mHqXyRRdQFIT0eD9JkK3jiCBHPowjKVdSJehLOW1X8Z1
mL6OT/+8h44QSEJlbwUqBO5rjPqDTosNp1fVAgMBAAGjggPlMIID4TAOBgNVHQ8B
Af8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIwADAdBgNV
HQ4EFgQUH/4fXjkxWtE5FmZ6IutDE1zVhUswHwYDVR0jBBgwFoAU3hse7XkV1D43
JMMhu+w0OW1CsjAwWAYIKwYBBQUHAQEETDBKMCEGCCsGAQUFBzABhhVodHRwOi8v
by5wa2kuZ29vZy93cjIwJQYIKwYBBQUHMAKGGWh0dHA6Ly9pLnBraS5nb29nL3dy
Mi5jcnQwggG9BgNVHREEggG0MIIBsIIRKi5nb29nbGV2aWRlby5jb22CFCouYTEu
Z29vZ2xldmlkZW8uY29tghYqLmExLmdvb2dsZXZhZHMtY24uY29tggkqLmJkbi5k
ZXaCFSoub3JpZ2luLXRlc3QuYmRuLmRldoInKi5jLmRvYy0wLTAtc2ouc2ouZ29v
Z2xldXNlcmNvbnRlbnQuY29tggwqLmMuMm1kbi5uZXSCDyouYy4ybWRuLWNuLm5l
dIIVKi5kYWkuZ29vZ2xldmlkZW8uY29tghcqLmRhaS5nb29nbGV2YWRzLWNuLmNv
bYIPKi5nb29nbGV6aXAubmV0ggoqLmd2dDEuY29tgg0qLmd2dDEtY24uY29tghcq
Lm9mZmxpbmUtbWFwcy5ndnQxLmNvbYIPKi5zbmFwLmd2dDEuY29tghIqLnNuYXAu
Z3Z0MS1jbi5jb22CESouZ2NwY2RuLmd2dDEuY29tghN4bi0tbmdzdHItbHJhOGou
Y29tghUqLnhuLS1uZ3N0ci1scmE4ai5jb22CC2kueXRpbWcuY29tgh1wbGF5LWxo
Lmdvb2dsZXVzZXJjb250ZW50LmNvbTATBgNVHSAEDDAKMAgGBmeBDAECATA2BgNV
HR8ELzAtMCugKaAnhiVodHRwOi8vYy5wa2kuZ29vZy93cjIvOVVWYk4wdzVFNlku
Y3JsMIIBAgYKKwYBBAHWeQIEAgSB8wSB8ADuAHUA2ra/az+1tiKfm8K7XGvocJFx
bLtRhIU0vaQ9MEjX+6sAAAGSbMFtswAABAMARjBEAiBvQgbBag/jILElGpUslp7A
oiwIfMyiESWEJcWHJL4wQQIgKmCB3OfC4TjewalH6D39QQdlEh5H4wfxQ43Euvc6
qAgAdQB2/4g/Crb7lVHCYcz1h7o0tKTNuyncaEIKn+ZnTFo6dAAAAZJswW1vAAAE
AwBGMEQCIBOaBzT7mL2XbrEulrmYWv5j259vAWYAN9QFRD5T7+mIAiByREv9q2fL
CafNH5O/l+X9NhSA4lnKnQ5A8opi+BmTIDANBgkqhkiG9w0BAQsFAAOCAQEAUYuo
5gI7WH9IYCeQmd7Xi2GvzloiPEJa1iCO30TQgax6Ip7tNFcFRh9TfHmqqHNR9ZVy
X5KMXFmFe4XhzEep0vi42NoxOkAG4sbqW9FbPaUUoTWj/vGmRvye70mnlxFutWls
JuNXGxrqaZLiX501BMgbESuM06VYlstISG1yB8EjT3h1cS4HFptUpozWw4lVMh7K
cMaP5X81/lFqp2fI5vu5CAEGX2Q51YeTdOxaj4Ege2XE8b7Ygd18Mh5wRleVOAul
RZRQMFlt+4WlCyTq5/5yVQyw7da1B6G75R4iapSQoHwa7guZXzzfZtPV+uD74A7P
q5t1uNfYvCrhtIrL9A==
-----END CERTIFICATE-----
subject=CN=*.googlevideo.com
issuer=C=US, O=Google Trust Services, CN=WR2
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 4918 bytes and written 402 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 2048 bit
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
^C
```
</details>

**Comment by syntezit** [comment]
> Похоже, я случайно обнаружил ещё один способ обхода блокировки `*.googlevideo.com`. Оказывается, некоторые сервера GGC корректно работают даже с неправильным SNI. Т. е. если тупо поменять SNI на `www.google.com` в ClientHello, сервер всё равно отвечает сертификатом `*.googlevideo.com`. Может кто-нибудь подтвердить?
> 
> Эксперимент
<details>
  <summary>Похоже на то</summary>
<pre>
openssl s_client -connect 74.125.163.97:443 -servername "www.google.com"
CONNECTED(00000003)
depth=2 C = US, O = Google Trust Services LLC, CN = GTS Root R1
verify return:1
depth=1 C = US, O = Google Trust Services, CN = WR2
verify return:1
depth=0 CN = *.googlevideo.com
verify return:1
---
Certificate chain
 0 s:CN = *.googlevideo.com
   i:C = US, O = Google Trust Services, CN = WR2
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Oct  8 14:31:06 2024 GMT; NotAfter: Dec 17 14:31:05 2024 GMT
 1 s:C = US, O = Google Trust Services, CN = WR2
   i:C = US, O = Google Trust Services LLC, CN = GTS Root R1
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Dec 13 09:00:00 2023 GMT; NotAfter: Feb 20 14:00:00 2029 GMT
 2 s:C = US, O = Google Trust Services LLC, CN = GTS Root R1
   i:C = BE, O = GlobalSign nv-sa, OU = Root CA, CN = GlobalSign Root CA
   a:PKEY: rsaEncryption, 4096 (bit); sigalg: RSA-SHA256
   v:NotBefore: Jun 19 00:00:42 2020 GMT; NotAfter: Jan 28 00:00:42 2028 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIGyTCCBbGgAwIBAgIRAIiaKa0lehC2CcuLTc82plUwDQYJKoZIhvcNAQELBQAw
OzELMAkGA1UEBhMCVVMxHjAcBgNVBAoTFUdvb2dsZSBUcnVzdCBTZXJ2aWNlczEM
MAoGA1UEAxMDV1IyMB4XDTI0MTAwODE0MzEwNloXDTI0MTIxNzE0MzEwNVowHDEa
MBgGA1UEAwwRKi5nb29nbGV2aWRlby5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IB
DwAwggEKAoIBAQCs9EuG9WWcDQ46fKeVBYW8yEbKrnEOtuZWkCnTm5WiszhH6Jp/
m2RcOCODRk29Xj7eoj3qDVOmKv/CNNnqtt5VF1p34r/REqpfQhYLWsLe5Ay2ODpB
7fg80akq9LRsomBfLQk5Pwc1L5G7JaqssILWlIdsAhxmAUAGydMgUs6HbLIO85FS
0TUhqZxGwDOrheyH5MXb0xmOf/q178A5Z90hhypDPH4SKR+9sYqL2DCSseF0oDzd
uACJkCdw1RPhGD2oOZ48mHqXyRRdQFIT0eD9JkK3jiCBHPowjKVdSJehLOW1X8Z1
mL6OT/+8h44QSEJlbwUqBO5rjPqDTosNp1fVAgMBAAGjggPlMIID4TAOBgNVHQ8B
Af8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIwADAdBgNV
HQ4EFgQUH/4fXjkxWtE5FmZ6IutDE1zVhUswHwYDVR0jBBgwFoAU3hse7XkV1D43
JMMhu+w0OW1CsjAwWAYIKwYBBQUHAQEETDBKMCEGCCsGAQUFBzABhhVodHRwOi8v
by5wa2kuZ29vZy93cjIwJQYIKwYBBQUHMAKGGWh0dHA6Ly9pLnBraS5nb29nL3dy
Mi5jcnQwggG9BgNVHREEggG0MIIBsIIRKi5nb29nbGV2aWRlby5jb22CFCouYTEu
Z29vZ2xldmlkZW8uY29tghYqLmExLmdvb2dsZXZhZHMtY24uY29tggkqLmJkbi5k
ZXaCFSoub3JpZ2luLXRlc3QuYmRuLmRldoInKi5jLmRvYy0wLTAtc2ouc2ouZ29v
Z2xldXNlcmNvbnRlbnQuY29tggwqLmMuMm1kbi5uZXSCDyouYy4ybWRuLWNuLm5l
dIIVKi5kYWkuZ29vZ2xldmlkZW8uY29tghcqLmRhaS5nb29nbGV2YWRzLWNuLmNv
bYIPKi5nb29nbGV6aXAubmV0ggoqLmd2dDEuY29tgg0qLmd2dDEtY24uY29tghcq
Lm9mZmxpbmUtbWFwcy5ndnQxLmNvbYIPKi5zbmFwLmd2dDEuY29tghIqLnNuYXAu
Z3Z0MS1jbi5jb22CESouZ2NwY2RuLmd2dDEuY29tghN4bi0tbmdzdHItbHJhOGou
Y29tghUqLnhuLS1uZ3N0ci1scmE4ai5jb22CC2kueXRpbWcuY29tgh1wbGF5LWxo
Lmdvb2dsZXVzZXJjb250ZW50LmNvbTATBgNVHSAEDDAKMAgGBmeBDAECATA2BgNV
HR8ELzAtMCugKaAnhiVodHRwOi8vYy5wa2kuZ29vZy93cjIvOVVWYk4wdzVFNlku
Y3JsMIIBAgYKKwYBBAHWeQIEAgSB8wSB8ADuAHUA2ra/az+1tiKfm8K7XGvocJFx
bLtRhIU0vaQ9MEjX+6sAAAGSbMFtswAABAMARjBEAiBvQgbBag/jILElGpUslp7A
oiwIfMyiESWEJcWHJL4wQQIgKmCB3OfC4TjewalH6D39QQdlEh5H4wfxQ43Euvc6
qAgAdQB2/4g/Crb7lVHCYcz1h7o0tKTNuyncaEIKn+ZnTFo6dAAAAZJswW1vAAAE
AwBGMEQCIBOaBzT7mL2XbrEulrmYWv5j259vAWYAN9QFRD5T7+mIAiByREv9q2fL
CafNH5O/l+X9NhSA4lnKnQ5A8opi+BmTIDANBgkqhkiG9w0BAQsFAAOCAQEAUYuo
5gI7WH9IYCeQmd7Xi2GvzloiPEJa1iCO30TQgax6Ip7tNFcFRh9TfHmqqHNR9ZVy
X5KMXFmFe4XhzEep0vi42NoxOkAG4sbqW9FbPaUUoTWj/vGmRvye70mnlxFutWls
JuNXGxrqaZLiX501BMgbESuM06VYlstISG1yB8EjT3h1cS4HFptUpozWw4lVMh7K
cMaP5X81/lFqp2fI5vu5CAEGX2Q51YeTdOxaj4Ege2XE8b7Ygd18Mh5wRleVOAul
RZRQMFlt+4WlCyTq5/5yVQyw7da1B6G75R4iapSQoHwa7guZXzzfZtPV+uD74A7P
q5t1uNfYvCrhtIrL9A==
-----END CERTIFICATE-----
subject=CN = *.googlevideo.com
issuer=C = US, O = Google Trust Services, CN = WR2
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 4918 bytes and written 400 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 2048 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
</pre>
</details>

**Comment by solodyagin** [comment]
Провайдер "МТС", работает заклинание `-s1 -q1 -Y -Ar -s5 -o1+s -At -f-1 -r1+s -As -s1 -o1 +s -s-1 -An`

**Comment by Wyrmlet** [comment]
> Провайдер "МТС", работает заклинание `-s1 -q1 -Y -Ar -s5 -o1+s -At -f-1 -r1+s -As -s1 -o1 +s -s-1 -An`

Вроде пашет на провайдере "Подряд", спасибо большое

**Comment by MrMeshok** [comment]
У меня стояло `-s2 -d2` поменял на `-s2 -d3` и заработало

**Comment by AntonyBenjiro** [comment]
МГТС
Пока работает с такими настройками:
`./ciadpi -d1 -o25+s`

**Comment by radioxoma** [comment]
## Двойные сплиты
@AntonyBenjiro, спасибо. Раньше на МГТС Youtube работал через: `ciadpi -d1 -r1`, теперь:

    ciadpi -d1 -o23+s  # -o в диапазоне 23-32  # Поломанные сайты возвращают RST, что отлично
    ciadpi -d1 -r23+s  # -r в диапазоне 23-32  # Вроде ломает меньше сайтов
    ciadpi -o1 -o23+s  # -o в диапазоне 23-32 (второй)

Если не стоит сверхзадачи разблокировать все сайты одной командой, то, в принципе, сойдёт `-o1 -o25+s -T3 -At`. Если какие-то сайты поломались (прислали RST или зависли с `-T3`), то  byedpi попробует переподключиться напрямую.

## Подмена sni

Чтобы фейковые `-f` пакеты дошли до DPI, но не дошли до сервера, нужно использовать `--md5sig`, `--ip-opt --no-ipv6` или подбирать `--ttl`. Метод ломает работу DoH.

### Linux

    ciadpi --tls-sni "www.google.com" -f-2 -r23+s --md5sig
    ciadpi --tls-sni "www.google.com" -f-2 -r23+s --ip-opt --no-ipv6  # --ip-opt с IPv6 не работало

@hufrea `--ip-opt` в 0.13+ не работает с IPv6 (или не понятно как правильно включить).

### Windows

В byedpi `--ttl 8` по умолчанию. Нужно подбирать для каждого провайдера. Диапазон 1-30 (?), использовать наименьшую рабочую цифру. В качестве альтернативы, в [README](https://github.com/hufrea/byedpi) есть рекомендация использовать `--disorder 1 --fake -1` вместо `--ttl`.

    ciadpi --tls-sni "www.google.com" -f-2 -r23+s --ttl 8  # --ttl нужно подбирать
    ciadpi --tls-sni "www.google.com" -d1 -f-1 -r6+s -r23+s  # Гипотеза, не проверял в работе

***

Проверять YouTube удобно при помощи большого видео, уже собранного на сервере в один файл `-fb`:

    yt-dlp --proxy socks5://localhost:1080 --no-cache-dir -fb https://youtu.be/Sagg08DrO5U -o /dev/null

**Reply by EvgenKo423** [reply]
[`--ip-opt` тоже только для линукса](https://github.com/hufrea/byedpi/blob/v0.15/main.c#L99-L102).

**Reply by kolserdav** [reply]
> ciadpi --tls-sni "www.google.com" -f-2 -r23+s --md5sig

Для Linux так сработало

**Reply by kolserdav** [reply]
> ciadpi --tls-sni "www.google.com" -f-2 -r23+s --ttl 8  # --ttl нужно подбирать

для Win 10 так сработало с 2

**Comment by Miraihi** [comment]
> Проверять удобно при помощи большого видео, уже собранного на сервере в один файл `-fb`:
> 
> ```
> yt-dlp --proxy socks5://localhost:1080 --no-cache-dir -fb https://youtu.be/Sagg08DrO5U -o /dev/null
> ```

Хорошая идея для теста. И да, раньше для Youtube вообще не требовалось модифицировать ServerHello. `-r` сделал огромную разницу после последних изменений в DPI.

**Comment by vehametra** [comment]
> > У меня на IPv6 работает так:
> > ```
> > ciadpi --hosts youtube.txt --fake -2 --tls-sni "www.google.com" --ip-opt --tlsrec=6+s --tlsrec 27+s
> > ```
> 
> Мне еще disorder понадобился, без него даже страницы не открывались. Получилось такое: `--fake -2 --disorder 2 --tlsrec=6+s --tlsrec 27+s`

`ciadpi --debug 1 --split 1 --disoob 3 --disorder 7 --fake -2 --tls-sni www.google.com --ip-opt --tlsrec=6+s  --fake 2 --tlsrec 27+s --split 2+s --tlsrec 3+s --mod-http d --disorder 19  --fake 3+s --mod-http h -F -H /root/hosts.txt -b 65536  `

у меня сейчас что-то такое.  и видео переодические обрывается. (прямо сейчас полёт нормальный, но утром было хуже) 3 дня назад представленный выше вариант с двумя tls-rec и disorder  работал идеально. Куда бы сюда вставить -A  опцию. Есть идееи об автоподборе параметров?

**Comment by Miraihi** [comment]
> > > У меня на IPv6 работает так:
> > > ```
> > > ciadpi --hosts youtube.txt --fake -2 --tls-sni "www.google.com" --ip-opt --tlsrec=6+s --tlsrec 27+s
> > > ```
> > 
> > 
> > Мне еще disorder понадобился, без него даже страницы не открывались. Получилось такое: `--fake -2 --disorder 2 --tlsrec=6+s --tlsrec 27+s`
> 
> `ciadpi --debug 1 --split 1 --disoob 3 --disorder 7 --fake -2 --tls-sni www.google.com --ip-opt --tlsrec=6+s --fake 2 --tlsrec 27+s --split 2+s --tlsrec 3+s --mod-http d --disorder 19 --fake 3+s --mod-http h -F -H /root/hosts.txt -b 65536 `
> 
> у меня сейчас что-то такое. и видео переодические обрывается. (прямо сейчас полёт нормальный, но утром было хуже) 3 дня назад представленный выше вариант с двумя tls-rec и disorder работал идеально. Куда бы сюда вставить -A опцию. Есть идееи об автоподборе параметров?

На моём опыте есть два рабочих варианта - с `disorder 1-2` и высоким (25+s и выше) `oob` (Представленный выше пользователем radioxima), но без `fake`, или с `disorder 1-2` и отрицательным fake но высоким `tlsrec` (25+s и выше). К fake желательно выставлять `--ttl`, настолько низкий насколько возможно (У меня 3).
Кстати, при общении с youtube можно полностью игнорировать протокол http, всё идет через https. (`--proto https,tls`)

**Comment by VijitCoder** [comment]
А может кто-нибудь засветить список доменов для  `--hosts`? Мне проще спросить, чем нагуглить или самому собирать )

**Comment by Miraihi** [comment]
> А может кто-нибудь засветить список доменов для `--hosts`? Мне проще спросить, чем нагуглить или самому собирать )

Для youtube у меня работает с этими:
```
youtube.com
youtu.be
yt.be
googlevideo.com
ytimg.com
ggpht.com
gvt1.com
youtube-nocookie.com
youtube-ui.l.google.com
youtubeembeddedplayer.googleapis.com
youtube.googleapis.com
youtubei.googleapis.com
yt-video-upload.l.google.com
wide-youtube.l.google.com
```

**Comment by karasjr** [comment]
Может кто подскажет как хром приручить? на лисе через прокси работает на ура, хром нет. Если что, кибер конечно же выключен. Ubuntu 24.04

**Reply by Ved-un** [reply]
Запустить с этими параметрами    
`google-chrome-stable --proxy-server=socks5://0.0.0.0:1080`

или прописать в дектоп файл вместо дефолтной строки запуска    
`Exec=google-chrome-stable --proxy-server=socks5://0.0.0.0:1080 ` 

При необходимости изменить в этой команде   
`google-chrome-stable` 
на любой другой ваш хромиум браузер

**Comment by pipipupu-ui** [comment]
> Уже несколько дней сам ютуб работает, но видео не загружаются. Возможно какая-то новая блокировка для `googlevideo.com`? Возможно дубликат [№190](https://github.com/hufrea/byedpi/issues/190) , но я не уверен.

Как я понял, блокируются все адреса где есть 
`google.com`

**Comment by Miraihi** [comment]
> Может кто подскажет как хром приручить? на лисе через прокси работает на ура, хром нет. Если что, кибер конечно же выключен. Ubuntu 24.04

Перво-наперво отключить протокол Quic.
Ну и конечно можно просто воспользоваться другим chromium браузером, например Brave. Вы ничего не теряете, но приобретаете встроенный adblock.

**Comment by Vortelio** [comment]
> Может кто подскажет как хром приручить? на лисе через прокси работает на ура, хром нет. Если что, кибер конечно же выключен. Ubuntu 24.04

Пользуюсь этим расширением 
https://github.com/txthinking/socks5-configurator
https://chromewebstore.google.com/detail/socks5-configurator/hnpgnjkeaobghpjjhaiemlahikgmnghb

**Comment by zimbabwe303** [comment]
Сегодня у меня FreeTube стал постоянно сбоить с ошибкой "[BAD_HTTP_STATUS: 403] Potential causes: IP block or streaming URL deciphering failed". Не знаю пока это Гугл мудрит с копиразмом или же ТСПУ подкрутили. Но фраза "IP block" настораживает, конечно. Строка настроек `-s1 -q1 -Y -At -T2 -b1000 -S -f-1 -r1+sm -As` работала без проблем с самой осени.

Попробую подкрутить что-нибудь, но чутьё подсказывает, что на этот раз это, скорее всего, Гугл, т.к. с моего прямого русского айпишника всё больше и больше не пускают на разные зарубежные сайты, дескать бывает, что от вас идёт кармически негативный трафик (а по факту, просто вынуждены соблюдать санкции, а то федералы закроют). В итоге давят нас со всех сторон...

Похоже, придётся полностью переползать на трёхбуквенную технологию, с риском начать получать от ТСПУ временные полные баны (в Китае уже давно так). Потом пойдут ухищрения с обфусцированными тунелями (которые, конечно, палятся, если через них гнать видеотрафик), ротацией IP (которая один фиг не помогает), балансировкой трафика между несколькими серверами, XHTTP и прочими и прочими страданиями молодого Вертера... А куда деваться? Информация-то на YouTube уникальная, все самые лучшие технические каналы англоязычные, привык уже к ним. Один Astrum чего стоит. А Electronics Repair School, Learn Electronics Repair, NorthridgeFix, DiodeGoneWild, Michael Dranfield, Shango066?

**Reply by zimbabwe303** [reply]
Вышел апдейт FreeTube для flatpak, просмотрел сейчас два довольно длинных видео, вроде не было больше ошибок. Старые настройки работают. Вернее, добавил туда -L1 (`-s1 -q1 -Y -At -T2 -L1 -b1000 -S -f-1 -r1+sm -As`): не знаю насколько оно нужно, просто иногда кэширование прекращается и качество падает до 360p; посмотрю, может поможет.

**Reply by rezawq** [reply]
Спасибо за настройку, работает пока что.

**Reply by zimbabwe303** [reply]
Сейчас тестирую немножко другой вариант: `-s1 -q15 -Y -At -T2 -b1000 -S -f-1 -r1+sm -As`

`-L1` не нужно, оно ничего не дало, вместо этого повысил `-q` с 1 до 15: есть гипотеза, что они в ТСПУ время от времени увеличивают размер анализируемого блока, но т.к. ТСПУ постоянно жрёт много электричества, этот размер им всё равно приходится держать в неких разумных пределах, поэтому задача сделать так, чтобы у нас точка сплита была за пределами этого размера. Многие люди, сидящие на Ростелекоме, упоминали, что им помогала сдвижка позиции сплита подальше (это параметры -s, -d, -o и -q), иной раз до каких-то очень далёких значений, чуть ни -s128, но из-за этого коннекты начинают сильно тормозить, конечно. Попробую, может и у моего местного провайдера что-то подобное теперь стало.
